### DISCUSSION TITLE
Trouble in SofaPython 3 binding

### DISCUSSION THREAD

**Discussion by ISIR-GuillaumeS** [sofa-framework/sofa#4052] [Build / Config / Environment] [answered]
Hi guys !

I would like to use SofaPython3 to make my Sofa scene communicate with a python script. I di dan in-tree build of SofaPython3 plugin but that was not sufficient to use Sofa in a python script (I am using VSCode). In the docs it is written to just do :

`plugin.SofaPython3/build $ cmake --install`

However, as you can see below I do not have any build rep in my SofaPython3 folder (nor inside the other folders).

![image](https://github.com/sofa-framework/sofa/assets/135211551/a188053a-d77e-4d91-8e8a-3b23d77264c4)

Do you have an idea why and how can I do the bindings to use Sofa in a python interpreter ?

Thanks !

**Comment by bakpaul** [comment, marked as answer]
If you made an in-tree compilation of sofaPython3 (using the cmake variable CMAKE_EXTERNAL_DIRECTORIES) the build files should be located in the sofa build directory itself. The library are located in the folder `$sofabuilddirectory$/lib/`. 
After locating your libraries path you could either install the libs by following the documentation, or you can add them to your python environment by setting your `PYTHONPATH` variable like this : `export PYTHONPATH=$PYTHONPATH:path/to/your/compiled/lib`

**Reply by ISIR-GuillaumeS** [reply]
Hi ! Thanks for replying ! Indeed, by adding to the environment variables things that were missing I achieve to make it works !
